irc/api.go

```diff
@@ -222,3 +228,179 @@ func (a *ergoAPI) handleAccountDetails(w http.ResponseWriter, r *http.Request) {
 
 	a.writeJSONResponse(response, w, r)
 }
+
+type apiNsInfoRequest struct {
+	Nick string `json:"nick"`
+}
+
+type apiNsInfoResponse struct {
+	apiGenericResponse
+	AccountName  string   `json:"accountName,omitempty"`
+	RegisteredAt string   `json:"registeredAt,omitempty"`
+	Channels     []string `json:"channels,omitempty"`
+	ChannelCount int      `json:"channelCount,omitempty"`
+}
+
+func (a *ergoAPI) handleNsInfo(w http.ResponseWriter, r *http.Request) {
+	var request apiNsInfoRequest
+	if err := a.decodeJSONRequest(&request, w, r); err != nil {
+		return
+	}
+
+	var response apiNsInfoResponse
+
+	if request.Nick != "" {
+		// Look up the account associated with this nick
+		accountName := a.server.accounts.NickToAccount(request.Nick)
+		if accountName == "" {
+			response.Success = false
+			a.writeJSONResponse(response, w, r)
+			return
+		}
+
+		// Load the account details
+		accountData, err := a.server.accounts.LoadAccount(accountName)
+		if err != nil {
+			response.Success = false
+			a.writeJSONResponse(response, w, r)
+			return
+		}
+
+		// Get the channels registered to this account
+		channels := a.server.channels.ChannelsForAccount(accountName)
+
+		// Populate the response
+		response.Success = true
+		response.AccountName = accountData.Name
+		response.RegisteredAt = accountData.RegisteredAt.Format("Mon, 02 Jan 2006 15:04:05 UTC")
+		response.Channels = channels
+		response.ChannelCount = len(channels)
+	} else {
+		response.Success = false
+		response.ErrorCode = "INVALID_REQUEST"
+	}
+
+	a.writeJSONResponse(response, w, r)
+}
+
+type apiAccountListRequest struct {
+	Limit  int    `json:"limit,omitempty"`
+	Offset int    `json:"offset,omitempty"`
+	Filter string `json:"filter,omitempty"`
+}
+
+type apiAccountListResponse struct {
+	Accounts   []apiAccountDetailsResponse `json:"accounts"`
+	TotalCount int                         `json:"totalCount"`
+}
+
+func (a *ergoAPI) handleAccountList(w http.ResponseWriter, r *http.Request) {
+	var request struct {
+		Limit  int    `json:"limit,omitempty"`
+		Filter string `json:"filter,omitempty"`
+	}
+	if err := a.decodeJSONRequest(&request, w, r); err != nil {
+		return
+	}
+
+	var response apiAccountListResponse
+	var accounts []string
+
+	// Get all account names
+	accountNamePrefix := fmt.Sprintf(keyAccountName, "")
+	a.server.store.View(func(tx *buntdb.Tx) error {
```

Comment:
We're trying to move away from application code interacting with buntdb directly --- this should be a method of `AccountManager`